src/reusableState/customList/StringListReducers.ts

```diff
@@ -0,0 +1,60 @@
+import * as _ from 'underscore';
+import {IItemBoxProps} from '../../components/itemBox/ItemBox';
+import {IReduxAction} from '../../utils/ReduxUtils';
+import {IStringListPayload, StringListActions} from './StringListActions';
+
+export type IStringListCompositeState = {[id: string]: IStringListState};
+
+export interface IStringListState {
+    id: string;
+    list: string[];
+}
+
+export const stringListInitialState: IStringListCompositeState = {};
+
+export const convertStringListToItemsBox = (list: string[], itemsBoxParams: Partial<IItemBoxProps> = {}): IItemBoxProps[] => _.map(list, (value: string) => ({...itemsBoxParams, value}));
+export const convertItemsBoxToStringList = (items: IItemBoxProps[]): string[] => _.pluck(items, 'value');
+
+export const stringListCompositeState = (
+    state: IStringListCompositeState = stringListInitialState,
+    action: IReduxAction<IStringListPayload>,
+): IStringListCompositeState => {
+
+    if (!action.payload || (!state[action.payload.id] && action.type !== StringListActions.add)) {
+        return {...state};
+    }
+
+    const stateList: IStringListState = {...state[action.payload.id]};
+    switch (action.type) {
+        case StringListActions.add:
+            return {
+                ...state,
+                [action.payload.id]: {
+                    id: action.payload.id,
+                    list: action.payload.list || [],
+                },
+            };
+        case StringListActions.remove:
+            return {
+                ..._.omit(state, [action.payload.id]),
+            };
+        case StringListActions.addValue:
+            return {
+                ...state,
+                [action.payload.id]: {
+                    ...stateList,

```

Comment:
i think you can avoid having a const `stateList` and destructure `{......state[action.payload.id]}`  

directly

Reply:
its too avoid any prob if we add other param in the futur. I create a new object to make sure to not use a reference

Reply:
it doesn't use a reference since you are destructuring state[action.payload.id] anyways, not sure to get your point maybe i'm missing something